immudb/handler/verifiedreference.py

```diff
@@ -32,32 +33,34 @@ def call(service: schema_pb2_grpc.ImmuServiceStub, rs: RootService, refkey: byte
         proveSinceTx=state.txId
     )
     vtx = service.VerifiableSetReference(vreq)
-    if vtx.tx.metadata.nentries != 1:
-        raise VerificationException
-    tx = immudb.store.TxFrom(vtx.tx)
-    inclusionProof = tx.Proof(immudb.store.EncodeKey(key))
-    ekv = immudb.store.EncodeReference(key, refkey, atTx)
-    verifies = immudb.store.VerifyInclusion(
-        inclusionProof, ekv.Digest(), tx.eh())
-    if not verifies:
-        raise VerificationException
-    if state.txId == 0:
-        sourceID = tx.ID
-        sourceAlh = tx.Alh
-    else:
-        sourceID = state.txId
-        sourceAlh = immudb.store.DigestFrom(state.txHash)
-    targetID = tx.ID
-    targetAlh = tx.Alh
-    verifies = immudb.store.VerifyDualProof(
-        immudb.htree.DualProofFrom(vtx.dualProof),
-        sourceID,
-        targetID,
-        sourceAlh,
-        targetAlh,
-    )
+    if vtx.tx.header.nentries != 1:
+        raise ErrCorruptedData
+    tx = schema.TxFromProto(vtx.tx)
+    entrySpecDigest = store.EntrySpecDigestFor(tx.header.version)
+    inclusionProof = tx.Proof(database.EncodeKey(key))
+    # TODO: Is metadata missing here?
```

Comment:
We're not setting metadata in the request (in line ~34) - thus the metadata should be empty. We should check though if it is empty before doing the proof. The proof will fail if server injected some metadata anyway but let's have a good error message here.

Reply:
hmm, the matching line in Golang is https://github.com/codenotary/immudb/blob/master/pkg/client/client.go#L1217 . The TODO was meant for clarification, if Golang SDK is doing things right here. Generally, I think we should be as close as possible to the Golang SDK.